components/channel_view/index.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017 Mattermost Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {get} from 'mattermost-redux/selectors/entities/preferences';
+
+import {getDirectTeammate} from 'utils/utils.jsx';
+import {Constants, TutorialSteps, Preferences} from 'utils/constants.jsx';
+
+import ChannelView from './channel_view.jsx';
+
+function mapStateToProps(state, ownProps) {
+    const channel = state.entities.channels.channels[state.entities.channels.currentChannelId];
+
+    let deactivatedChannel = false;
+    if (channel && channel.type === Constants.DM_CHANNEL) {
```

Comment:
Maybe factor this out into a crude selector? getDirectTeammate isn't redux-friendly without a complete rewrite, but you could hack something out without rewriting it like
```
const getDeactivatedChannel = createSelector(
    (state) => state.users.users, // This isn't used, but it'll cause it to recalculate if any users change
    (state, channelId) => channelId,
    (users, channelId) => {
        return getDirectTeammate(channelId);
    }
);
```

Reply:
If I'm going to do that, I'd rather just do `getDirectTeammate` properly as well but that's all kind of out of the scope of this PR. If you think the current implementation is going to cause issues or you feel strongly about it I'll make the update but otherwise that can come in a future PR when `getDirectTeammate` is re-implemented

Reply:
I think we need to do something to keep it from recalculating this every time the store changes. I agree that rewriting `getDirectTeammate` doesn't need to be part of this PR, that's why I was suggesting that hacky one as an interim solution.

Reply:
Ok, I'll use your selector then 👍